### PR TITLE
implement max lengths for input dialogs that are sent to the server

### PR DESCRIPTION
**Pull request by ebbit1q**
## Short roundup of the initial problem
people can enter really long messages where they shouldn't ever need that much

## What will change with this Pull Request?
- add a shared header that contains some constants
- add these constants as a max value to every dialog with info that is sent to the server
- some refactors (this always happens doesn't it)

### CODE REVIEW

File: cockatrice/src/player.cpp [hunk 1]
```diff
@@ -3141,7 +3147,8 @@ void Player::actSetAnnotation()
     bool ok;
     dialogSemaphore = true;
     QString annotation = QInputDialog::getMultiLineText(game, tr("Set annotation"),
-                                                        tr("Please enter the new annotation:"), oldAnnotation, &ok);
+                                                        tr("Please enter the new annotation:"), oldAnnotation, &ok)
+                             .left(MAX_NAME_LENGTH);
```
Comment (ZeldaZach) [inline cockatrice/src/player.cpp:3151]: Is this what clangify does? Weird

Comment (ebbit1q) [reply cockatrice/src/player.cpp:3151]: yes, there is no proper way to validate this type of widget so I just slapped left on it

**Review by ZeldaZach — APPROVED** [review]
Seems fine

File: cockatrice/src/dlg_edit_avatar.cpp [hunk 2]
```diff
@@ -75,6 +77,13 @@ QByteArray DlgEditAvatar::getImage()
     QByteArray ba;
     QBuffer buffer(&ba);
     buffer.open(QIODevice::WriteOnly);
-    image.save(&buffer, "JPG");
-    return ba;
+    for (;;) {
```
Comment (ZeldaZach) [inline cockatrice/src/dlg_edit_avatar.cpp:80]: any reason you do `for (;;)` instead of `while (true)`?

Comment (ebbit1q) [reply cockatrice/src/dlg_edit_avatar.cpp:80]: I just prefer it, I think it looks nice

**Pull request merged**